skbio/sequence/_sequence.py

```diff
@@ -187,30 +351,107 @@ def metadata(self):
     def metadata(self, metadata):
         if not isinstance(metadata, dict):
             raise TypeError("metadata must be a dict")
-        self._metadata = metadata
+        # shallow copy
+        self._metadata = metadata.copy()
+
+    @metadata.deleter
+    def metadata(self):
+        if hasattr(self, '_metadata'):
+            del self._metadata
+        self.metadata = {}
 
     @property
     def positional_metadata(self):
-        """Pandas DataFrame containing metadata on a per-character basis.
+        """pandas.DataFrame containing metadata on a per-character basis.
```

Comment:
[Here](https://github.com/biocore/scikit-bio/pull/954/files#diff-4ac32a78649ca5bdd8e0ba38b7006a1eR13) you use pd.DataFrame instead of pandas.DataFrame. Should probably be consistent.